### PR TITLE
db layer added

### PR DESCRIPTION
**Pull request by kashifkhan0771**
Added db layer for micro service.

### CODE REVIEW

File: db/client.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		log.Fatalf("Datastore factory %s does not exist.", name)
+		return
+	}
+	_, ok := datastoreFactories[name]
+	if ok {
+		log.Fatalf("Datastore factory %s already registered. Ignoring.", name)
+		return
+	}
+	datastoreFactories[name] = factory
+}
```
Comment (ahab94) [inline db/client.go:39]: Add line ending

Comment (kashifkhan0771) [reply db/client.go:39]: okay

**Review by ahab94 — CHANGES_REQUESTED** [review]
Feedback

File: Dockerfile.dev [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      jq -r '.assets[] | select(.name | contains("'"$(uname | tr '[:upper:]' '[:lower:]')"'_amd64")) | .browser_download_url') \
+      && curl -o /usr/local/bin/swagger -L'#' "$download_url" \
+      && chmod +x /usr/local/bin/swagger
+
+RUN go get github.com/golangci/golangci-lint/cmd/golangci-lint \
+    && go get golang.org/x/tools/cmd/goimports
+
+RUN CGO_ENABLED=0
+
+ARG DEVELOPER
```
Comment (ahab94) [inline Dockerfile.dev:23]: Add line ending

File: Makefile.variables [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+DOCKRUN := docker run --rm \
+	-v ${ROOT}:/${PROJECT}/src/${IMPORT_PATH} \
+	${DEV_IMAGE}
+
+DOCKTEST := docker run --rm \
+    -v ${ROOT}:/${PROJECT}/src/${IMPORT_PATH} \
+    --link host-service-mongo-db:host-service-mongo-db \
+    --link host-service-mysql-db:host-service-mysql-db \
+    ${DEV_IMAGE}
```
Comment (ahab94) [inline Makefile.variables:17]: Add linr ending

File: Dockerfile.dev (hunk 2 above)
Comment (kashifkhan0771) [reply Dockerfile.dev:23]: okay

File: db/mongo/client.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+
+	"github.com/kashifkhan0771/HostService/config"
```
Comment (ahab94) [inline db/mongo/client.go:8]: Imports are not organised

File: db/mysql/client.go [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
+	"github.com/kashifkhan0771/HostService/config"
+	"github.com/kashifkhan0771/HostService/db"
+	domainErr "github.com/kashifkhan0771/HostService/errors"
+	"github.com/kashifkhan0771/HostService/models"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+	"github.com/spf13/viper"
```
Comment (ahab94) [inline db/mysql/client.go:16]: Imports are not ordered

File: db/mysql/client_test.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
+package mysql
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"os"
+	"reflect"
```
Comment (ahab94) [inline db/mysql/client_test.go:9]: Imports are not ordered

File: host_service.sql [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+USE host_service;
+
+CREATE TABLE hosts(
+    id       VARCHAR(255),
+    ip       TEXT,
+    name     TEXT,
+    metadata VARCHAR(255),
+    status   VARCHAR(255)
+);
```
Comment (ahab94) [inline host_service.sql:11]: Add line ending

**Comment by kashifkhan0771** [comment]
Okay i will try to resolve all of these ASAP.

File: scripts/check.sh [hunk 8]
```diff
@@ -1 +1,2 @@
+#!/bin/bash
 golangci-lint run
```
Comment (ahab94) [inline scripts/check.sh:2]: Add line ending

File: scripts/format.sh [hunk 9]
```diff
@@ -1 +1,2 @@
+#!/bin/bash
 gofmt -l -w .
```
Comment (ahab94) [inline scripts/format.sh:2]: Add line ending

File: scripts/test.sh [hunk 10]
```diff
@@ -1,3 +1,2 @@
-echo "Testing The Models"
-#cd ../models || exit
-go test ./models
\ No newline at end of file
+#!/bin/bash
+go test ./...
```
Comment (ahab94) [inline scripts/test.sh:2]: Add line ending

**Pull request merged**